### ISSUE TITLE
Adding a bridge immediately reboots CCU

### ISSUE THREAD

**Issue by marvingrieger** [j-a-n/homematic-addon-hue#14]
I just updated the HomeMatic-Addon to the newest version. Then, I needed to pair the bridge again.

When pairing the bridge, all lights blink shortly and then the CCU (running RaspberryMatic) immediately reboots.

I use an external log-server, however, it seems like the nothing relevant gets logged (even when setting the log-level of the CCU and the plugin to verbose).

Anyone experienced something similar?

**Comment by marvingrieger** [comment]
I also performed a "tail -f" on the "/tmp/hue-addon-log.txt" - nothing suspicious here. The last entry is something like:

`[0] [2019-11-06 21:04:57] [2519] api request: XXX - PUT - /api/XXX/lights/28/state - {"alert":"select"}
[0] [2019-11-06 21:04:57] [2519] api response: [{"success":{"/lights/28/state/alert":"select"}}]`

Also, stopping the CUXD-Daemon does not prevent the reboot.

**Comment by j-a-n** [comment]
Not sure what could cause the reboot.
Please try version 2.1.14 with the following change:
After establishing a link alert select is send to group 0 instead of every individual light.

**Comment by marvingrieger** [comment]
Just tried the newest version, the problem still exists. Now, there is no further entry in the log files after the lights blink.

Is there some kind of development version with even more logging to see what's going on?

**Comment by marvingrieger** [comment]
I identified the problem. There seems to be a bug if the name of the bridge contains a dash like "The-Bridge". When renaming it to "TheBridge", it works as expected.

May I suggest to only close the issue when it is really resolved?

**Comment by j-a-n** [comment]
You can reopen an issue if it is not resolved.

**Comment by marvingrieger** [comment]
Sorry, I don't know how. I guess you're seeing a "reopen" button, I'm not.

**Comment by j-a-n** [comment]
Could you please try if version 2.1.15 fixes the issue?

**Comment by marvingrieger** [comment]
Just tested the new version, unfortunately the bug still exists.

**Comment by j-a-n** [comment]
2.1.16?

**Comment by marvingrieger** [comment]
Nope, still rebooting. Maybe provide some version with really verbose logging on a development branch?

**Comment by j-a-n** [comment]
Could you please provide me the output of
`/usr/local/addons/hue/hue.tcl <bridge-id> request GET config`
If you don't want to post it here, you can send it to oss@janschneider.net

**Comment by marvingrieger** [comment]
I would do that. The problem is, there is no configured bridge. When performing an update to the newest version, the pairing was lost. The devices are there, but the corresponding bridge is not. That's why I'm seeing all these errors in the log:

`[2] [2019-11-08 19:04:26] [1292] Failed to update XXX light 28: Hue bridge XXX not configured - NONE - Hue bridge XXX not configured while executing "error "Hue bridge ${bridge_id} not configured"" (procedure "request" line 6) invoked from within "request "status" $bridge_id "GET" $obj_path" (procedure "hue::get_object_state" line 3) invoked from within "hue::get_object_state $bridge_id "${obj}s/${num}"" (procedure "update_cuxd_device" line 8) invoked from within "update_cuxd_device [lindex $tmp 0] [lindex $tmp 1] [lindex $tmp 2]"`

However, when pairing / adding the bridge again (after renaming it), all devices are immediately linked and work as expected.

Or do you want me to rename the bridge and pair it again?

If you want to chat, you can reach me on Skype (marvin_grieger).

**Comment by j-a-n** [comment]
okay, are you able to edit a file on the ccu?

**Comment by marvingrieger** [comment]
Sure, no problem

**Comment by j-a-n** [comment]
okay, please add the write_log in /usr/local/addons/hue/www/rest.cgi around line 155.
`
set config [hue::api_request "info" $ip 80 $username "GET" "config"]
hue::write_log 1 $config
regexp {\"name\"\s*:\s*\"([^\"]+)\"} $config match name
`
Set the log level to error or higher.
Then try again to establish a connection to the bridge.
Now the bridge config gets logged.

**Comment by marvingrieger** [comment]
I injected the line as proposed:

<img width="835" alt="Code" src="https://user-images.githubusercontent.com/1997765/68510327-80958a80-0273-11ea-890d-d4eb25b5e482.png">

However, this change does not result in any logging information, even when restarting the hue daemon etc.

Also, the behavior seems to be non-deterministic. In some instances, I was able to add the bridge, in other instances the reboot is performed.

When the reboot is performed, it happens after this "add" button is clicked:

<img width="945" alt="Bildschirmfoto 2019-11-08 um 21 54 31" src="https://user-images.githubusercontent.com/1997765/68510454-e41fb800-0273-11ea-8c15-bd1abef769d8.png">
